compiler/parser/RdrHsSyn.hs

```
@@ -2284,12 +2289,18 @@ validateInterface name requires decls = do
     (e:es) -> do
       mapM_ (\e' -> report (getLoc e') "Multiple 'ensure' declarations") es
       pure (Just e)
+  viViewType <- case ibdViewDecls of
+    [] -> pure Nothing
```

Do we nod need to report an error here similar to how we report an error for a missign signatory declaration?

---

I used to perform this check in the parser, but moved it into the typechecker's check phase, rather than the parser, so that we can merge this change without breaking all of the existing interface tests.

If we did this check here, we would need to change all of the tests to have a placeholder viewtype decl. Then, later, in the PR that actually generates view methods during LF conversion, we would need to change the same tests again to have matching view methods in implementing templates. Or, we could merge the test changes along with LF conversion but then we'd end up with two fairly big PRs that are tightly coupled.

---

fair point, maybe add a TODO to fix that once you’ve adjusted all tests?

---

Done